src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -586,20 +583,18 @@ private void CheckDisposed()
 
         private static bool IsWebSocketsSupported()
         {
-#if NETCOREAPP3_0
+#if NETSTANDARD2_1
             // .NET Core 2.1 and above has a managed implementation
             return true;
 #else
-            var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
-            if (!isWindows)
+            try
             {
-                // Assume other OSes have websockets
+                new ClientWebSocket().Dispose();
                 return true;
             }
-            else
+            catch
```

Comment:
Do we know what get thrown in practice?

Reply:
`NotSupportedException`

Reply:
I figured. This is gross, but catching only the `NotSupportedException` is probably even worse.

Reply:
Do we cache this now?